index.js

```diff
@@ -14,26 +30,36 @@ module.exports = (template, data) => {
 	const doubleBraceRegex = /{{(\d+|[a-z$_][a-z\d$_]*?(?:\.[a-z\d$_]*?)*?)}}/gi;
 
 	if (doubleBraceRegex.test(template)) {
-		template = template.replace(doubleBraceRegex, (_, key) => {
-			let result = data;
+		template = template.replace(doubleBraceRegex, (placeholder, key) => {
+			const value = key.split('.').reduce((value, key) => value ? value[key] : undefined, data);
```

Comment:
I prefer to not use `reduce`. A `for-of` loop is usually better.

Reply:
But it requires a reassignment of the variable, which is generally considered harmful these days. Why do you prefer that?

Reply:
> which is generally considered harmful these days

Global mutations are indeed bad. Localized mutations are totally fine.